src/core/libmaven/jsonReports.cpp

```diff
@@ -1,286 +1,678 @@
+                               _mavenParameters->filterline, _mavenParameters->amuQ1, _mavenParameters->amuQ3);
+        }
+       else {//MS1 case
+            //TODO: same problem here: need the ppm that was used, or the slice object
+            //for mz could rely on same computation being done way above
+            //redoing it only for code clarity
+
+            MassCutoff *massCutoff=_mavenParameters->compoundMassCutoffWindow;
+            slice->mzmin = slice->mz - massCutoff->massCutoffValue(slice->mz);
+            slice->mzmax = slice->mz + massCutoff->massCutoffValue(slice->mz);
+            slice->rtmin = grp.minRt - _outputRtWindow;
+            slice->rtmax = grp.maxRt + _outputRtWindow;
+            eic = (it)->getEIC(slice->mzmin,slice->mzmax,slice->rtmin,slice->rtmax,1,
+                                _mavenParameters->eicType,
+                                _mavenParameters->filterline);
+            }
+    }
+
+    else {//no compound information
+          //TODO: same problem here: need the ppm that was used, or the slice object
+        slice->mz=grp.meanMz;
+        MassCutoff *massCutoff=_mavenParameters->compoundMassCutoffWindow;
+        slice->mzmin = slice->mz - massCutoff->massCutoffValue(slice->mz);
+        slice->mzmax = slice->mz + massCutoff->massCutoffValue(slice->mz);
+        slice->rtmin = grp.minRt - _outputRtWindow;
+        slice->rtmax = grp.maxRt + _outputRtWindow;
+        eic = (it)->getEIC(slice->mzmin,slice->mzmax,slice->rtmin,slice->rtmax,1,
+                            _mavenParameters->eicType,
+                            _mavenParameters->filterline);
+    }
+
+    //TODO: for MS1 we've already limited RT range, but for MS/MS the entire RT range of the SRM will be output
+    //either check here or edit getEIC functionality
+    if(eic) {
+        
+        int N = eic->rt.size();
+        myfile << ",\n" << "\"eic\": {" ;
+        myfile << "\"rt\": [";
+
+        for(int i=0;i<N;i++){
+            if ( eic->rt[i] > 0) {
+                myfile << eic->rt[i];
+                if (i < N - 1) myfile << ",";
+            }
+        }
+            
+        myfile << "],\n" ; //rt
+        myfile << "\"intensity\": [";
+        for(int i=0;i<N;i++){
+            if ( eic->rt[i] > 0) {
+                myfile << eic->intensity[i];
+                if (i < N - 1) myfile << ",";
+            }
+        }
+            
+        myfile << "]" ; //intensity
+        myfile << "\n}" ;//eic
+
+        myfile << "\n}" ; //peak
+        delete(eic);
+    }
+   
+}
+
+
+
+void JSONReports::save(string filename,vector<PeakGroup> allgroups,vector<mzSample*> samples) {
+    ofstream myfile(filename.c_str());
+    myfile << setprecision(10);
+
+    myfile << "{\"groups\": [" <<endl;
+
+    int groupId=0;
+    int metaGroupId=0;
+   
+    for(int i=0; i < static_cast<int>(allgroups.size()) ; i++ ) {

```

Comment:
Instead of casting vector size to be an integer, you can init your index (`i`), as a size type (`size_t`).